services/can_logger/can_logger.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+import socket
+import postgres
+import csv
+import os
+
+def csv_init(host_interface):
+
+    log = open('/data/log/' + host_interface + '.csv', mode = 'a') 
+
+    return log
+
+def write_to_csv(timestamp, can_id, can_data, log):
+
+        log = csv.writer(log, delimiter = ',', quotechar = '"',
+                         quoting = csv.QUOTE_MINIMAL)
+
+        log.writerow([timestamp, can_id, can_data])
```

Comment:
Still not a huge fan of how it's being done. I _think_ this will work without modification of the main function. This will keep us from creating a new `csv.writer` object every time `write_to_csv()` is called


```suggestion
def csv_init(host_interface):
    # Open the log file
    logfd = open('/data/log/' + host_interface + '.csv', mode = 'a') 
    # Make csv writer object using log file
    logcsv = csv.writer(log, delimiter = ',', quotechar = '"',
                         quoting = csv.QUOTE_MINIMAL)
    return logcsv

def write_to_csv(timestamp, can_id, can_data, log):
        log.writerow([timestamp, can_id, can_data])
```

Also for some reason I feel like the `log` object should be the first argument. Not sure why, maybe a old C convention that the pass by reference arguments are usually first? @abalmos might have an opinion

Reply:
You mean, writing to CSV as a funcion instead of using the code in the main function? I thought it would be easier to understand if the code inside the loop has less amount of lines.